import_craft.py

```diff
@@ -0,0 +1,291 @@
+            index += 1
+
+
+def rename_materials(ob):
+    select_children(ob)
+    index = 0
+    for obj in bpy.context.selected_objects:
+        for mat in obj.material_slots:
+            if not mat.material.name.split('_')[-1] == 'material':
+                mat.material.name = '{}{}_material'.format(obj.name, index)
+                index += 1
+
+
+def rename_images(ob, directory):
+    index = 0
+    for img in bpy.data.images:
+        imgname = img.name
+        if not imgname.split('_')[-1] == 'image':
+            img.name = ob.name + imgname + '_image'
+            img.name = '{}{}_image'.format(ob.name, index)
+            index += 1
+
+
+def select_children(obj):
+    bpy.context.scene.objects.active = obj
+    obj.select = True
+    bpy.ops.object.select_grouped(extend=True)
+
+
+def unselect_all_objects():
+    for obj in bpy.context.selected_objects:
+        obj.select = False
+
+
+def duplicate_object_hierarchy(obj):
+    unselect_all_objects()
+    select_children(obj)
+    bpy.ops.object.duplicate(linked=True)
+    return bpy.context.scene.objects.active
+
+
+def remove_object_list(array):
+    unselect_all_objects()
+    for ob in array:
+        select_children(ob)
+    bpy.ops.object.delete()
+
+
+def read_vector(value):
+    # Unity is Y-Up, blender is Z-up
+    values_str = value.split(',')
+    return (float(values_str[0]), float(values_str[2]), float(values_str[1]))
+
+
+def read_quaternion(value):
+    # Unity is xyzw, blender is wxyz. However, Unity is left-handed and
+    # blender is right handed. To convert between LH and RH (either
+    # direction), just swap y and z and reverse the rotation direction.
+    quat_str = value.split(',')
+    return (float(quat_str[3]), -float(quat_str[0]), -float(quat_str[2]), -float(quat_str[1]))
+
+
+class CraftReader(object):
+    def __init__(self):
+        self.ship_name = ""
+        # Set of required models
+        self.prefabs = []
+        self.nb_total_parts = 0
+
+    def report(self, type_message, message):
+        print("[{}] {}".format(','.join(type_message), message))
+
+    def rename_data_elements(self, ob, directory):
+        ''' Rename the datas to avoid overriding while loading parts '''
+        unselect_all_objects()
+        rename_materials(ob)
+        rename_textures(ob)
+        rename_images(ob, directory)
+
+    def smooth_object_meshes(self, ob):
+        select_children(ob)
+        children = bpy.context.selected_objects
+        unselect_all_objects()
+        for child in children:
+            if child.type == "MESH":
+                bpy.context.scene.objects.active = child
+                bpy.ops.object.mode_set(mode='EDIT')
+                bpy.ops.mesh.faces_shade_smooth()
+                # Need to come back to OBJECT mode to avoid context error
+                bpy.ops.object.mode_set(mode='OBJECT')
+
+    def read_parts_models(self, partsdict, colliders):
+        for part in partsdict:
```

Comment:
this loop is really hard to read; it loops on dict keys but never actually uses the `key` but only the associated `value`.
```
for part in partsdict.values():
    unselect_all_objects()
    result = import_mu.import_mu(self, bpy.context, part['mu'], colliders) 
```
is more readable and more efficient.